### ISSUE TITLE
manage command import geojson able to do twice

### ISSUE THREAD

**Issue by stijnvanhoey** [inbo/vespa-watch#84]
I got the follow error when redploying the app:

```
[2018-11-19T16:38:52.927Z] INFO  [32261] - [Application update app-181119_173718@3/AppDeployStage0/EbExtensionPostBuild/Infra-EmbeddedPostBuild/postbuild_0_vespawatch/Command 05_get_observations_inaturalist] : Activity execution failed, because: Will import observations from project #22865...
  Step 1: Will remove from our database all observations that originates from iNaturalist (they'll be recreated right after)...OK
  Step 2: Will parse the 274 matching observations reveived from iNaturalist.
  Now processing iNaturalist observation #1591681...Observation initially comes from a regular iNaturalist user. Creating a local observation for it. Traceback (most recent call last):
  File "manage.py", line 24, in <module>
  execute_from_command_line(sys.argv)
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
  utility.execute()
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
  self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/core/management/base.py", line 316, in run_from_argv
  self.execute(*args, **cmd_options)
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/core/management/base.py", line 353, in execute
  output = self.handle(*args, **options)
  File "/opt/python/bundle/3/app/vespawatch/management/commands/sync_pull.py", line 71, in handle
  r = create_observation_from_inat_data(inat_observation_data)
  File "/opt/python/bundle/3/app/vespawatch/models.py", line 120, in create_observation_from_inat_data
  observation_time=observation_time)  # TODO: What to do with iNat observations without (parsable) time?
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
  return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/db/models/query.py", line 413, in create
  obj.save(force_insert=True, using=self.db)
  File "/opt/python/bundle/3/app/vespawatch/models.py", line 367, in save
  self.auto_assign_zone()
  File "/opt/python/bundle/3/app/vespawatch/models.py", line 245, in auto_assign_zone
  self.zone = get_zone_for_coordinates(self.latitude, self.longitude)
  File "/opt/python/bundle/3/app/vespawatch/models.py", line 201, in get_zone_for_coordinates
  return FirefightersZone.objects.get(mpolygon__intersects=point)
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/db/models/manager.py", line 82, in manager_method
  return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/opt/python/run/venv/local/lib/python3.6/site-packages/django/db/models/query.py", line 403, in get
  (self.model._meta.object_name, num)
  vespawatch.models.MultipleObjectsReturned: get() returned more than one FirefightersZone -- it returned 2!
   (ElasticBeanstalk::ExternalInvocationError)
```

Due to the fact that I did twice `command:  "python manage.py import_firefighters_zones data/Brandweerzones_2019.geojson"` during subsequent deployments. I know this is an edge-case, but @niconoe  how should I solve this best? Should I add the `truncate` option to the command or should I adapt the code of the `import_firefighters_zones.py` script?

(I solved something similar on the [firefighters user accounts](https://github.com/inbo/vespa-watch/commit/b4d560d739c4c374218f349ed67d892ab44f25f4))

**Comment by niconoe** [comment]
Yep, those scripts were initially created to be run only once, on the initial deployment. 

Does ElasticBeans allows you to run a different set of commands for an initial deployment (creating everything from scratch) than for subsequent ones (only for application upgrade, so most of the time only updating the code itself + installing new requirements if necessary + running migrate + restarting app servers). That would be the cleanest way to avoid the issue I think, and that will probably be needed anyway....

Technically, the `--truncate` option will solve this on an (almost) empty database, but it will create issues with real data, because other models will refer to those (to be deleted) zones.

**Comment by stijnvanhoey** [comment]
Until now, the distinction between commands only on first deployment versus commands on each redeployment  was not required, as all functions required rerun or had no issue running again (the superuser command had the check included already).  I found a workaround [here](https://www.martin-brennan.com/run-aws-elastic-beanstalk-config-commands/) and I'll try to add that to the deployment, so we do not need command-specific requirements for the deployment.

**Comment by niconoe** [comment]
Thanks,

The changes that you did on the other scripts (only create if doesn't exists) can of course be done there too if this helps. It just wasn't my first suggestion because I had the feeling that the "first/subsequent" deployment distinction at the infrastructure would be needed soon anyway (maybe just because I'm use to having that for other projects). 

Just let me know of your investigation results!

**Comment by stijnvanhoey** [comment]
FYI, as we are not working in a single instance environment, but load balanced, servers are started/stopped when required (or during rolling updates). From the [documentation](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/ebextensions.html):

> The other sections of a configuration file (packages, sources, files, users, groups, commands, container_commands, and services) let you configure the EC2 instances that are launched in your environment. **Whenever a server is launched in your environment, Elastic Beanstalk runs the operations defined in these sections** to prepare the operating system and storage system for your application.

**Comment by stijnvanhoey** [comment]
just to remember

> Leader-only container commands are only executed during environment creation and deployments, while other commands and server customization operations are performed every time an instance is provisioned or updated.

Hence, leader-only is also not an option as this will be repeated when doing new deployments.

An option would be to document these commands as part of the general setup (cfr. S3 bucket creation) and not include it to the configurations for deployment.

**Comment by stijnvanhoey** [comment]
The distinction is made as following:

- first deployment only commands are provided using the `eb ssh` command line element, which is part of the setup, but not part of the `.config` container or machine commands
- each deployment commands (e.g. migrate) are provided within the `.ebextensions` configuration files.
